json_rpc/clients/httpclient.nim

```diff
@@ -45,16 +52,29 @@ method call*(client: RpcHttpClient, name: string,
     req = HttpClientRequestRef.post(client.httpSession,
                                     client.httpAddress.get,
                                     body = reqBody.toOpenArrayByte(0, reqBody.len - 1))
-    res = await req.send()
+    res =
+      try:
+        await req.send()
+      except CatchableError as exc:
+        error "Failed to send POST Request", message = exc.msg
+        raise exc
```

Comment:
By using this scheme you are leaking exception from lower level to upper level. Consumer of `json-rpc` should only obtain `JsonRpcError` but not `AsyncStreamError` or `HttpError`.

Reply:
68a879060869b489adab80a49e78fc7bcb595533